src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralService.java

```diff
@@ -61,6 +63,12 @@ public void handlePendingDeletionReferral(final OffenderPendingDeletion event) {
                         .offenderNumber(new OffenderNumber(event.getOffenderIdDisplay()))
                         .offenceCodes(event.getOffenceCodes())
                         .alertCodes(event.getAlertCodes())
+                        .bookingNos(event.getOffenderAliases().stream().flatMap(offenderAlias -> offenderAlias.getOffenderBookings().stream()).map(OffenderBooking::getBookingNo).collect(Collectors.toSet()))
```

Comment:
Might be nice to move this stream into its own method to help readability?

Reply:
Updated